py/server/deephaven/table.py

```diff
@@ -2123,7 +2167,7 @@ def select_distinct(self, formulas: Union[str, Sequence[str]] = None) -> Partiti
         """
         try:
             formulas = to_sequence(formulas)
-            with auto_locking_ctx(self):
+            with auto_locking_ctx(self), _query_scope_ctx():
```

Comment:
Did we remove formulas from joins, aggs, etc., or do they need a scope?

Reply:
I think so. Support formulas in joins, etc. adds minimum convenience but would be outweighed by the added complexity in the UI and in the query engine.